back/app/services/xlsx_export/sheet_generator.rb

```diff
@@ -0,0 +1,205 @@
+                  options[:style] = date_time
+                end
+                row.add_cell(value, options)
+              end
+            end
+          end
+        end
+      end
+    end
+
+    private
+
+    attr_reader :inputs, :fields_in_form, :participation_method, :include_private_attributes
+
+    def not_supported?(field)
+      field.code == 'idea_images_attributes' # Not supported by XlsxService
+    end
+
+    def input_id_report_field
+      ComputedFieldForReport.new(column_header_for('input_id'), ->(input) { input.id })
+    end
+
+    def author_name_report_field
+      ComputedFieldForReport.new(column_header_for('author_fullname'), ->(input) { input.author_name })
+    end
+
+    def author_email_report_field
+      ComputedFieldForReport.new(column_header_for('author_email'), ->(input) { input.author.email })
+    end
+
+    def author_id_report_field
+      ComputedFieldForReport.new(column_header_for('author_id'), ->(input) { input.author_id })
+    end
+
+    def latitude_report_field
+      ComputedFieldForReport.new(column_header_for('latitude'), ->(input) { input.location_point&.coordinates&.last })
+    end
+
+    def longitude_report_field
+      ComputedFieldForReport.new(column_header_for('longitude'), ->(input) { input.location_point&.coordinates&.first })
+    end
+
+    def created_at_report_field
+      ComputedFieldForReport.new(column_header_for('created_at'), ->(input) { input.created_at })
+    end
+
+    def published_at_report_field
+      ComputedFieldForReport.new(column_header_for('published_at'), ->(input) { input.published_at })
+    end
+
+    def comments_count_report_field
+      ComputedFieldForReport.new(column_header_for('comments_count'), ->(input) { input.comments_count })
+    end
+
+    def upvotes_count_report_field
+      ComputedFieldForReport.new(column_header_for('upvotes_count'), ->(input) { input.upvotes_count })
+    end
+
+    def downvotes_count_report_field
+      ComputedFieldForReport.new(column_header_for('downvotes_count'), ->(input) { input.downvotes_count })
+    end
+
+    def baskets_count_report_field
+      ComputedFieldForReport.new(column_header_for('baskets_count'), ->(input) { input.baskets_count })
+    end
+
+    def input_url_report_field
+      ComputedFieldForReport.new(
+        column_header_for('input_url'),
+        ->(input) { Frontend::UrlService.new.model_to_url(input) }
+      )
+    end
+
+    def project_report_field
+      ComputedFieldForReport.new(
+        column_header_for('project'),
+        ->(input) { MultilocService.new.t(input.project.title_multiloc) }
+      )
+    end
+
+    def status_report_field
+      ComputedFieldForReport.new(
+        column_header_for('status'),
+        ->(input) { MultilocService.new.t(input.idea_status&.title_multiloc) }
+      )
+    end
+
+    def assignee_fullname_report_field
+      ComputedFieldForReport.new(column_header_for('assignee_fullname'), ->(input) { input.assignee.full_name })
```

Comment:
I think the assignee is optional and therefore this code might give an error.

Reply:
Good catch.